unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,264 @@
+        repo = self.connection_config.get_repo()
+
+        # Load the Git tree with all files, and then create Ingest docs
+        # for all blobs, i.e. all files, ignoring directories
+        sha = self.connection_config.branch or repo.default_branch
+        git_tree = repo.get_git_tree(sha, recursive=True)
+
+        for element in git_tree.tree:
+            rel_path = element.path.replace(self.connection_config.repo_path, "").lstrip("/")
+            if element.type == "blob" and self.is_file_type_supported(element.path):
+                record_locator = {
+                    "repo_path": self.connection_config.repo_path,
+                    "file_path": element.path,
+                }
+                if self.connection_config.branch is not None:
+                    record_locator["branch"] = self.connection_config.branch
+
+                date_modified = datetime.strptime(
+                    element._headers["last-modified"],
+                    "%a, %d %b %Y %H:%M:%S %Z",
+                ).isoformat()
+
+                date_created = datetime.strptime(
+                    element._headers["date"],
+                    "%a, %d %b %Y %H:%M:%S %Z",
+                ).isoformat()
+
+                yield FileData(
+                    identifier=element.sha,
+                    connector_type=CONNECTOR_TYPE,
+                    source_identifiers=SourceIdentifiers(
+                        fullpath=element.path,
+                        filename=element.path.split("/")[-1],
+                        rel_path=rel_path,
+                    ),
+                    metadata=FileDataSourceMetadata(
+                        url=element.url,
+                        version=element.etag,
+                        record_locator=record_locator,
+                        date_modified=date_modified,
+                        date_created=date_created,
+                        date_processed=str(time()),
+                    ),
+                    additional_metadata={
+                        "content-type": element._headers.get("content-type", ""),
+                        "mode": element._rawData.get("mode", ""),
+                        "type": element._rawData.get("type", ""),
+                        "size": element._rawData.get("size", ""),
+                    },
+                )
+
+
+class GitHubDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitHubDownloader(Downloader):
+    connection_config: GitHubConnectionConfig
+    download_config: GitHubDownloaderConfig
+
+    @requires_dependencies(["github"], extras="github")
+    def _fetch_file(self, path):
+        from github.GithubException import UnknownObjectException
+
+        try:
+            content_file = self.connection_config.get_repo().get_contents(path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exists {self.connection_config.url}/{path}")
+            return None
```

Comment:
Let's reraise this as an error rather than returning None, I don't see any reason why we this error should be thrown without us being notified of it.